### PR TITLE
Implement Log4ShellExploiter

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #3388 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] ~Do all end-to-end tests pass?~
* [ ] ~Any other testing performed?~
    > ~Tested by {Running the Monkey locally with relevant config/running Island/...}~
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Unit tests fail due to a circular import.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
```
monkey/agent_plugins/exploiters/log4shell/src/log4shell_exploiter.py:49:5: C901 'Log4ShellExploiter.exploit_host' is too complex (11)
    def exploit_host(
    ^
1     C901 'Log4ShellExploiter.exploit_host' is too complex (11)
```

Are you running your pre-commit hooks?

**Comment by shreyamalviya** [comment]
> Unit tests fail due to a circular import

This'll go away once the exploit client is in place. I made temporary changes to verify that the tests pass.

> Are you running your pre-commit hooks?

Yes, I plan on getting to that.

File: monkey/agent_plugins/exploiters/log4shell/src/log4shell_exploiter.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+
+def _stop_agent_binary_http_server(
+    agent_binary_http_server: HTTPBytesServer, agent_binary_download_timeout: float
+):
+    try:
+        logger.debug("Stopping the Agent binary server")
+        agent_binary_http_server.stop(agent_binary_download_timeout)
+    except Exception:
+        logger.exception("An unexpected error occurred while stopping the HTTP server")
```
Comment (shreyamalviya) [inline monkey/agent_plugins/exploiters/log4shell/src/log4shell_exploiter.py:244]: Use `SERVER_SHUTDOWN_TIMEOUT`.

File: monkey/agent_plugins/exploiters/log4shell/src/log4shell_utils/service_exploiters/logstash.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from logging import getLogger
+
+import requests
+
+from common.common_consts.timeouts import MEDIUM_REQUEST_TIMEOUT
+from infection_monkey.exploit.log4shell_utils.service_exploiters import IServiceExploiter
```
Comment (ilija-lazoroski) [inline monkey/agent_plugins/exploiters/log4shell/src/log4shell_utils/service_exploiters/logstash.py:6]: We need to import this from the plugin itself

**Review by mssalvatore — CHANGES_REQUESTED** [review]
We should get rid of the `log4shell_utils` subdirectory. That was useful when everything was crammed together in `infection_monkey/exploits`, but since the plugin is now isolated, there's no clear differentiator between what belogs in `log4shell_utils` and what belongs at the top level. The distinction is meaningless.

**Pull request merged**